stdout-mq.js

```diff
@@ -197,12 +197,35 @@ if (!configOptions.spawnProcess) {
   });
   console.log(`Running child process with PID "${child.pid}"`);
 
+  let childProcessWasTerminated = false;
+
+  /**
+   * Close the main process when a child process was terminated
+   * @param {Function} nextCheckAfter Returns when should check a child process
+   * @param {number} maxSeconds How many seconds it should wait for a terminated child process
+   * @param {number} seconds Initial seconds to wait for a terminated child process
+   * @returns {void}
+   */
+  const closeMainProcess = function closeMainProcess(nextCheckAfter, maxSeconds, seconds) {
+    setTimeout(() => {
+      const shouldTerminateEvenIfChildProcessStillRunning = seconds >= maxSeconds;
+
+      if (childProcessWasTerminated || shouldTerminateEvenIfChildProcessStillRunning) {
+        process.exit(34);
+      } else {
+        closeMainProcess(nextCheckAfter, maxSeconds, nextCheckAfter(seconds));
```

Comment:
Instead of doing those awkward things with recursion just use events! 
1. Start listening child.on('exit'
2. Create timer for SIGKILL
3. Send signal to child process
4. When  child.on('exit' fires - remove timer & exit program